extensions/Lily/Data.js

```diff
@@ -0,0 +1,653 @@
+                value: "<",
+              },
+            ],
+          },
+          orderType: {
+            acceptReporters: false,
+            items: [
+              {
+                text: "reversed",
+                value: "reversed",
+              },
+              {
+                text: "ascending",
+                value: "ascending",
+              },
+              {
+                text: "descending",
+                value: "descending",
+              },
+              {
+                text: "randomised",
+                value: "randomised",
+              },
+            ],
+          },
+          orderTypeSort: {
+            acceptReporters: false,
+            items: [
+              {
+                text: "ascending",
+                value: "ascending",
+              },
+              {
+                text: "descending",
+                value: "descending",
+              },
+            ],
+          },
+          indexType: {
+            acceptReporters: false,
+            items: [
+              {
+                text: "first",
+                value: "first",
+              },
+              {
+                text: "last",
+                value: "last",
+              },
+              {
+                text: "random",
+                value: "random",
+              },
+            ],
+          },
+          lists: {
+            acceptReporters: true,
+            items: "_getLists",
+          },
+        },
+      };
+    }
+
+    deleteItems(args, util) {
+      const list = getVarObjectFromName(args.LIST, util, "list");
+      if (!list) return false;
+      const listLength = list.value.length;
+      let num1 = 0;
+      let num2 = 0;
+      if (!list) return;
+      if (args.NUM1 > args.NUM2) {
+        num1 = args.NUM2 - 1;
+        num2 = args.NUM1 - 1;
+      } else {
+        num1 = args.NUM1 - 1;
+        num2 = args.NUM2 - 1;
+      }
+      const listPart1 = list.value.slice(0, num1);
+      const listPart2 = list.value.slice(num2 + 1, listLength);
+      list.value = listPart1.concat(listPart2);
+      list._monitorUpToDate = false;
+    }
+
+    deleteAllOfItem(args, util) {
+      const list = getVarObjectFromName(args.LIST, util, "list");
+      if (!list) return;
+      const newList = list.value.filter(function (model) {
+        return model !== args.ITEM;
+      });
+      list.value = newList;
+      list._monitorUpToDate = false;
+    }
+
+    replaceAllOfItem(args, util) {
+      const list = getVarObjectFromName(args.LIST, util, "list");
+      if (!list) return;
+      const listLength = list.value.length;
+      const item1 = args.ITEM1;
+      const item2 = args.ITEM2;
+      let newList = [];
+      for (let i = 0; i < listLength; i++) {
+        if (list.value[i] === item1) {
+          newList.push(item2);
+        } else {
+          newList.push(list.value[i]);
+        }
+      }
+      list.value = newList;
+      list._monitorUpToDate = false;
```

Comment:
you are setting entirely new list so _monitorUpToDate not needed, strictly speaking